Tavisca.Bootcamp.LanguageBasics.Exercise1/Program.cs

```diff
@@ -14,16 +14,50 @@ static void Main(string[] args)
             Console.ReadKey(true);
         }
 
-        private static void Test(string args, int expected)
-        {
+        private static void Test(string args, int expected)
+        {
             var result = FindDigit(args).Equals(expected) ? "PASS" : "FAIL";
-            Console.WriteLine($"{args} : {result}");
+            Console.WriteLine($"{args} : {result}");
+        }
+
+        public static int FindDigit(string equation)
+        {
+            string[] arrABC = equation.Split("*");
+            string A = arrABC[0];
+            string[] arrBC =  arrABC[1].Split("=");
+            string B = arrBC[0];
+            string C = arrBC[1];
+
+            if(C.Contains('?'))
+                return CalValue(A, B, C, 3);
+            else if(A.Contains('?'))
+                return CalValue(C, B, A, 1);
+            else if(B.Contains('?'))
+                return CalValue(C, A, B, 2);
+            return -1;
         }
+        public static int CalValue(string X, string Y, string Z, int i)

```

Comment:
Follow SOLID programming principles that makes your code more readable.
Write meaningful method names, use enums to define cases, don't use int i and values 1,2,3 for cases.

https://itnext.io/solid-principles-explanation-and-examples-715b975dcad4
Use Singleresponsibillity principle in your code for this example.

Follow pesudocode : 
GetMissingDigit(string multiplier1, string multiplier2, string result)
{
  // Call appropriate method as required.
  
  int multiplier1;
  if(int.TryParse(multiplier1,out multiplier1) == false)
  {
	GetMissingDigitInMultiplier(int, string, int);
  }
  elseif(int.TryParse(multiplier1,out multiplier1) == false)
  {
   }
  else
  {
  GetMissingDigitInResult(int,int, string);
  }
}

GetMissingDigitInMultiplier(int multiplier1, string multiplier2, int result)
{
}

GetMissingDigitInResult(int multiplier1, int multiplier2, string result)
{
}

Reply:
Changed the design.